src/utils/gaa.js

```diff
@@ -2108,47 +2134,45 @@ export class GaaMetering {
             'userState.registrationTimestamp invalid, userState.registrationTimestamp needs to be an integer and in seconds'
           );
 
-          return false;
-        }
-
-        if (newUserState.registrationTimestamp > Date.now() / 1000) {
-          debugLog('userState.registrationTimestamp is in the future');
-
-          return false;
-        }
-
-        if (
-          newUserState.grantReason === GrantReasonType.SUBSCRIBER &&
-          !('subscriptionTimestamp' in newUserState)
+          noIssues = false;
+        } else if (
+          convertPotentialTimestampToSeconds(
+            newUserState.registrationTimestamp
+          ) >
+          Date.now() / 1000
         ) {
-          debugLog(
-            'subscriptionTimestamp is required if userState.grantReason is SUBSCRIBER'
-          );
-
-          return false;
-        }
-
-        if (
-          'subscriptionTimestamp' in newUserState &&
-          !(
-            typeof newUserState.subscriptionTimestamp === 'number' &&
-            newUserState.subscriptionTimestamp % 1 === 0
-          )
-        ) {
-          debugLog(
-            'userState.subscriptionTimestamp invalid, userState.subscriptionTimestamp needs to be an integer and in seconds'
-          );
+          debugLog('userState.registrationTimestamp is in the future');
 
-          return false;
+          noIssues = false;
         }
 
-        if (
-          'subscriptionTimestamp' in newUserState &&
-          newUserState.subscriptionTimestamp > Date.now() / 1000
-        ) {
-          debugLog('userState.subscriptionTimestamp is in the future');
-
-          return false;
+        if (newUserState.grantReason === GrantReasonType.SUBSCRIBER) {
+          if (!('subscriptionTimestamp' in newUserState)) {
+            debugLog(
+              'subscriptionTimestamp is required if userState.grantReason is SUBSCRIBER'
+            );
+
+            noIssues = false;
+          } else if (
+            !(
+              typeof newUserState.subscriptionTimestamp === 'number' &&
+              newUserState.subscriptionTimestamp % 1 === 0
```

Comment:
You could probably replace both of these with `Number.isInteger(newUserState.subscriptionTimestamp)`. Took me a second to grok what this was meant to read as

Reply:
I tried this but got a lint error:

```
2153:23  error  no ES2015 "Number" methods and properties allowed to be used  swg-internal/no-es2015-number-props
```